backend/src/main/java/com/huellapositiva/application/controller/HandlerPasswordApiController.java

```diff
@@ -76,4 +81,38 @@ public void sendEmailRecovery(@RequestParam("email") String email){
     public void changePassword(@PathVariable("hash") String hash, @RequestParam("newPassword") String password){
         credentialsAction.executePasswordChanging(hash, password);
     }
+
+    @Operation(
+            summary = "Update the password from profile",
+            description = "The user access his profile to update his password.",
+            tags = {"contactPerson, volunteers, profile"}
+    )
+    @ApiResponses(
+            value = {
+                    @ApiResponse(
+                            responseCode = "204",
+                            description = "No content, password updated and confirmation email sent successfully."
+                    ),
+                    @ApiResponse(
+                            responseCode = "409",
+                            description = "Conflict, the old password does not match or the new password is invalid."
+                    ),
+                    @ApiResponse(
+                            responseCode = "500",
+                            description = "Internal server error, could not fetch the user data due to a connectivity issue."
+                    )
+            }
+    )
+    @RolesAllowed({"VOLUNTEER", "CONTACT_PERSON"})
+    @PostMapping("/editPassword")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void editProfilePassword(@RequestParam("newPassword") String newPassword,
```

Comment:
We should encourage the use of the HTTP body to transfer information whenever possible. So it is better to model a DTO instead of using the request parameters.

Reply:
Good morning.

I understand your point of view, but the profile is not finished yet, it would be possible to return the user's profile information instead of a No_Content. Which for now is not possible because that endpoint is not done yet. I have it written to refactor that.

Reply:
The response status is unrelated to the fact that we should use the HTTP body to transmit data.

My comment is not for the response body but for the request body. Do not use request parameters to receive the newPassword, etc, use a dto instead and send it as a json body to the backend

Reply:
Ok, i understand, i must to use a dto.